### PR TITLE
[CI][ROCm] Mark 3 nondeterministic gfx950 models flaky to unblock inductor-periodic

### PR DESCRIPTION
**Pull request by bobrenjc93**
Unblocks the gfx950 half of #192862. `inductor-periodic` has been red on main
for 23 consecutive runs, last green 2026-08-07, so the workflow currently cannot
surface any real inductor regression.

Not an inductor change. `bb96371605b` (#188429, "Switch rocm-preview to TheRock
wheels and upgrade CI to 7.14") landed between the last green run and the first
red one. Despite being scoped to `rocm-preview` by title it modified
`.ci/docker/common/install_rocm.sh` and `.ci/docker/ubuntu-rocm/Dockerfile`, so
the shared `pytorch-linux-jammy-rocm-n-py3-benchmarks` image was rebuilt and
every ROCm job picked up the new stack (`BUILD_ROCM_VERSION` 7.2 -> 7.14, ROCm
root moving from `/opt/rocm` to the TheRock wheel).

## Why flaky_models rather than rebaselining

The first version of this PR pinned expected values per @naromero77amd's
guidance in #192862. **I ran it against the real gfx950 jobs and it does not
work.** Two of four jobs were fixed; two stayed red, both on the same mode:

```
dynamic_inductor_timm 2/2        vit_base_patch14_dinov2  IMPROVED: accuracy=pass, expected=fail_accuracy
dynamic_inductor_torchbench 2/2  torch_multimodal_clip    IMPROVED: accuracy=pass, expected=fail_accuracy
```

That invalidates the premise I gave for the pins:

- I described `torch_multimodal_clip` as failing 100% consistently and
  one-directionally, so a genuine numerical change. It **passed** on the first
  run after being pinned to `fail_accuracy`. The 9/9 sample was real but not
  predictive.
- I only changed `vit_base_patch14_dinov2` on the inference baselines, since
  only those were failing. The untouched **training** baseline then went red on
  an improvement.

`check_accuracy.py` fails a job on a mismatch in *either* direction, so pinning
a value for a nondeterministic model only changes which outcome turns the job
red. `flaky_models` is the mechanism built for this: membership yields
`PASS_BUT_FLAKY` / `FAIL_BUT_FLAKY`, and neither appends to `failed` or
`improved`. The ROCm block already carries `alexnet` and `demucs` for precisely
this root cause -- *"Discovered on gfx950 CI after ROCm 7.2 upgrade, eager mode
non determinism"*.

So this reverts the four CSV edits and adds the three models to the
ROCm-scoped set. Nothing changes for CUDA.

## Tradeoff

This suppresses accuracy signal for these three models on ROCm entirely, so a
genuine regression in them would no longer be caught. That is the same tradeoff
already accepted for `alexnet`/`demucs`, and #192862 stays open to investigate
the nondeterminism -- consistent with "we will go back and triage later".

If you would rather keep real signal for `torch_multimodal_clip` specifically,
the alternative is extending `get_eager_nondeterministic_models()` to cover the
ROCm paths (it currently returns entries only for two CUDA files). That tolerates
only `pass -> eager_two_runs_differ` while still catching a genuine
`fail_accuracy`. Happy to switch if preferred -- it is narrower but has no
existing ROCm precedent.

## Test plan

Every observed outcome simulated through the real `check_accuracy()` with the
reverted CSV values:

```
observed=pass                   job_fails=False
observed=eager_two_runs_differ  job_fails=False
observed=fail_accuracy          job_fails=False
```

Under the pinned approach two of those three turn the job red.

`ciflow/inductor-periodic` is applied so the gfx950 jobs actually run on this PR
-- note `ciflow/trunk` and `ciflow/inductor` do **not** trigger this workflow,
which is why the first revision went unverified. Final validation is those four
jobs going green here.

Authored with the assistance of an AI assistant.

## Maintainer decision

@naromero77amd confirmed in #192862 that ROCm prefers the scoped `flaky_models` approach. The alternative eager-nondeterministic handling is therefore not part of this change. The tracking issue remains open for root-causing and eventually removing these exemptions.

## Final validation

The `ciflow/inductor-periodic` branch run completed with all ROCm/gfx950 jobs green: https://github.com/pytorch/pytorch/actions/runs/31715044756. Its sole failure was an unrelated CPU Hugging Face offline-cache miss in `dynamic_cpu_inductor_timm 2/2`.

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @jataylo @hongxiayang @naromero77amd @pragupta @jerrymannil @xinyazhang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @azahed98

### CODE REVIEW

**Comment by bobrenjc93** [comment]
@naromero77amd @jeffdaily @jataylo — the requested validation is complete: the current PR checks are fully green, including every ROCm job. This is ready for review and merge; #192862 remains open to track the underlying gfx950 nondeterminism and eventual removal of the exemptions. Could one of you take a look?

(Follow-up posted by Codex.)

**Review by naromero77amd — APPROVED** [review]

**Comment by bobrenjc93** [comment]
@jansel, could you take a quick look and approve this? The ROCm behavior has been validated and approved by @naromero77amd, and all checks are green, but the merge bot requires an approval from an eligible Core Reviewer or Metamate. Once that approval lands, I will rerun `@pytorchbot merge`. #192862 will remain open to track the underlying gfx950 nondeterminism and eventual removal of these exemptions.

(Follow-up posted by Codex.)

**Review by jansel — APPROVED** [review]
